pkg/objectstore/objectstore_test.go

```diff
@@ -447,15 +450,27 @@ func (s *ObjectStoreProviderSuite) TestBucketGetRegions(c *C) {
 	}
 }
 
-func getSecret(c *C, osType ProviderType) *Secret {
+func getSecret(c *C, ctx context.Context, osType ProviderType) *Secret {
 	secret := &Secret{}
 	switch osType {
 	case ProviderTypeS3:
 		secret.Type = SecretTypeAwsAccessKey
+		awsAccessKeyID := os.Getenv("AWS_ACCESS_KEY_ID")
+		awsSecretAccessKey := os.Getenv("AWS_SECRET_ACCESS_KEY")
+		var awsSessionToken string
+		if role, ok := os.LookupEnv(aws.ConfigRole); ok {
+			creds, err := aws.SwitchRole(ctx, awsAccessKeyID, awsSecretAccessKey, role, assumeRoleDuration)
+			c.Check(err, IsNil)
+			val, err := creds.Get()
+			c.Check(err, IsNil)
+			awsAccessKeyID = val.AccessKeyID
+			awsSecretAccessKey = val.SecretAccessKey
+			awsSessionToken = val.SessionToken
+		}
```

Comment:
@tdmanv  With these changes, We run objectstore tests with AWS IAM role, one of the 8 tests fails with my credentials:
```
go test

----------------------------------------------------------------------
FAIL: objectstore_test.go:423: ObjectStoreProviderSuite.TestBucketGetRegions

objectstore_test.go:429:
    c.Check(err, IsNil)
... value *errors.withStack = Containers, listing the buckets: AccessDenied: Access Denied
	status code: 403, request id: FC39088EE1A52416, host id: g9XaZ2ITdRbK4rDjjR+MCkv8sGlS8lLvubXbq4jnOZRyB/CQU7e0Gk/hyKW86wvffTo4EcQjO4M= ("Containers, listing the buckets: AccessDenied: Access Denied\n\tstatus code: 403, request id: FC39088EE1A52416, host id: g9XaZ2ITdRbK4rDjjR+MCkv8sGlS8lLvubXbq4jnOZRyB/CQU7e0Gk/hyKW86wvffTo4EcQjO4M=")

objectstore_test.go:439:
    c.Check(err, IsNil)
... value *errors.withStack = Containers, listing the buckets: AccessDenied: Access Denied
	status code: 403, request id: B02BD08131EADD44, host id: UTVEUGHfLkhv6ZUs07OHvc8D74LSMMT4sw5ZBQecbjkqVtf55jteT3+TSytkhxgQ1yHGBe6LNK4= ("Containers, listing the buckets: AccessDenied: Access Denied\n\tstatus code: 403, request id: B02BD08131EADD44, host id: UTVEUGHfLkhv6ZUs07OHvc8D74LSMMT4sw5ZBQecbjkqVtf55jteT3+TSytkhxgQ1yHGBe6LNK4=")

objectstore_test.go:439:
    c.Check(err, IsNil)
... value *errors.withStack = Containers, listing the buckets: AccessDenied: Access Denied
	status code: 403, request id: A34CE00D4BC213E7, host id: BtgIfwhLgvf+WCZDOvi6jG8zg63XVaHp1f4jUn8UL0qUMHyuFDTJhqqB4wXoiwnooF0NSy90zfE= ("Containers, listing the buckets: AccessDenied: Access Denied\n\tstatus code: 403, request id: A34CE00D4BC213E7, host id: BtgIfwhLgvf+WCZDOvi6jG8zg63XVaHp1f4jUn8UL0qUMHyuFDTJhqqB4wXoiwnooF0NSy90zfE=")

objectstore_test.go:439:
    c.Check(err, IsNil)
... value *errors.withStack = Containers, listing the buckets: AccessDenied: Access Denied
	status code: 403, request id: 9017B3A3E89EA391, host id: 6+JXjooSmq1FMlitzW787EsZSz8ktC3DWfvC64wRvsvBhrgvkKi9ikCgi2PpqR4KbeblH6KAq4w= ("Containers, listing the buckets: AccessDenied: Access Denied\n\tstatus code: 403, request id: 9017B3A3E89EA391, host id: 6+JXjooSmq1FMlitzW787EsZSz8ktC3DWfvC64wRvsvBhrgvkKi9ikCgi2PpqR4KbeblH6KAq4w=")

objectstore_test.go:439:
    c.Check(err, IsNil)
... value *errors.withStack = Containers, listing the buckets: AccessDenied: Access Denied
	status code: 403, request id: 9B88820B15FB881A, host id: 8oyRbM9NywaXjgQYISX3j1n+cTS3okFPb32nMt3fpjKXNH8TzwzcjfbQVYEj9DuNHQLwIWmZPjQ= ("Containers, listing the buckets: AccessDenied: Access Denied\n\tstatus code: 403, request id: 9B88820B15FB881A, host id: 8oyRbM9NywaXjgQYISX3j1n+cTS3okFPb32nMt3fpjKXNH8TzwzcjfbQVYEj9DuNHQLwIWmZPjQ=")

OOPS: 7 passed, 1 skipped, 1 FAILED
--- FAIL: Test (8.70s)
FAIL
exit status 1
FAIL	github.com/kanisterio/kanister/pkg/objectstore	8.729s
```

Reply:
@SupriyaKasten is this failing before or after we get temporary credentials?

Reply:
These fail after we get the temp crews, although 7 of the tests pass with the temp credentials. The call to `s.provider.ListBuckets(ctx)` fails with these temp credentials.

Reply:
If so, then this is an issue with the Role we're changing to in these tests, so won't this fail if we run this test with anyone's creds?

Reply:
Discussed offline with @tdmanv and @vkamra, skipping the failing test since we want to run the tests with available role permissions(our s3 role does not have list bucket permission)